assets/stylesheets/kitten/components/form/_search-input.scss

```diff
@@ -62,60 +62,74 @@
   // Border radius Text - input
   $submit-border-radius: k-px-to-rem(4px);
 
+  // Tiny height from k-TextInput
+  $tiny-height: k-px-to-rem(40px);
+
   // Transition
   $transition-speed: .2s;
 
   .k-SearchInput {
-    position: relative;
-    display: inline-block;
+    display: table;
+    position: relative; // to place an absolute unit
+
+    border-collapse: separate;
+    box-sizing: border-box;
+    width: k-px-to-rem(160px) + $submit-horizontal-padding * 2 + $icon-size;
   }
 
   .k-SearchInput__input {
-    width: k-px-to-rem(160px);
+    display: table-cell;
+    vertical-align: middle;
+
+    border-top-right-radius: 0 !important;
+    border-bottom-right-radius: 0 !important;
+    border-right: 0 !important;
+    box-sizing: border-box;
+    width: initial !important;
     padding-right: $input-right-padding;
 
     &:focus ~ .k-SearchInput__submit {
-      border-left-color: $border-focus;
+      border-color: $border-focus;
     }
 
     // You can add this class in JavaScript to manage `k-SearchInput__input`
     // hover style depending on a following sibling hover state.
-    &.is-hover {
+    &.is-hover ~ .k-SearchInput__submit {
```

Comment:
`k-TextInput` est `.k-SearchInput__input`. Là le `.is-hover` est sur `.k-SearchInput__input`.
Mais je ne suis pas sûr de bien comprendre ta question, enfin où tu veux en venir :).